version: "3.9"

services:
  nginx:
    image: nginx:latest
    depends_on:
      - django
    volumes:
      - ${PWD}/homework_3/nginx/nginx.conf:/etc/nginx/nginx.conf
    ports:
      - "8080:80"
    environment:
      - NGINX_PORT=80

  django:
    build:
      context: ./distance_learning_system
    container_name: app_homework_3
    depends_on:
      db:
        condition: service_healthy
    volumes:
      - ${PWD}/homework_3/distance_learning_system:/app
    ports:
      - "8000:8000"
    restart: unless-stopped
    entrypoint: python manage.py runserver 0.0.0.0:8000

  db:
    hostname: postgres
    image: postgres:latest
    ports:
      - "5432:5432"
    restart: always
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=dls_db
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    healthcheck:
      test: [ "CMD-SHELL", "echo 'SELECT 1;' | psql -U postgres -d dls_db" ]
      interval: 3s
      timeout: 5s
      retries: 10

volumes:
  postgres_data:
    driver: local